com.unity.netcode.gameobjects/Runtime/Spawning/NetworkPrefabHandler.cs

```diff
@@ -223,6 +266,43 @@ public bool RemoveHandler(uint globalObjectIdHash)
         /// <returns>true or false</returns>
         internal bool ContainsHandler(uint networkPrefabHash) => m_PrefabAssetToPrefabHandler.ContainsKey(networkPrefabHash) || m_PrefabInstanceToPrefabAsset.ContainsKey(networkPrefabHash);
 
+        /// <summary>
+        /// Returns the <see cref="INetworkPrefabInstanceHandlerWithData"/> implementation for a given <see cref="NetworkObject.GlobalObjectIdHash"/>
+        /// </summary>
+        /// <param name="objectHash"></param>
+        /// <param name="handler"></param>
+        /// <returns></returns>
+        internal bool TryGetHandlerWithData(uint objectHash, out INetworkPrefabInstanceHandlerWithData handler)
+        {
+            return m_PrefabAssetToPrefabHandlerWithData.TryGetValue(objectHash, out handler);
+        }
+
+        /// <summary>
+        /// Reads the instantiation data for a given <see cref="NetworkObject.GlobalObjectIdHash"/>
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="objectHash"></param>
+        /// <param name="serializer"></param>
+        internal FastBufferReader GetInstantiationDataReader(uint objectHash, FastBufferReader fastBufferReader)
+        {
+            if (!TryGetHandlerWithData(objectHash, out var _))
+            {
+                if (NetworkManager.Singleton.LogLevel <= LogLevel.Developer)
+                {
+                    Debug.LogWarning($"No handler with data found for object hash {objectHash}.");
+                }
+                return default;
+            }
+
+            fastBufferReader.ReadValueSafe(out int dataSize);
```

Comment:
Ah, I was too slow with my review here. After some testing and poking on my side I agree with your initial implementation!

Reply:
No problem! Just to confirm, should I revert to the original version with the buffer copy?
```csharp
var result = new FastBufferReader(fastBufferReader, Collections.Allocator.Temp, dataSize, dataStartPos);
```
Or do you prefer keeping the optimized unsafe version which, as I understand, avoids the allocation entirely?